sdk/src/Services/DynamoDBv2/Custom/DataModel/Context.cs

```diff
@@ -53,6 +53,61 @@ public partial class DynamoDBContext : IDynamoDBContext
 
         #endregion
 
+        #region Public methods
+        /// <summary>
+        /// Adds a <see cref="Table"/> to this context's internal cache, which
+        /// will avoid the need to fetch table metadata automatically from DynamoDB.
+        /// This may be used in conjunction with an <see cref="ITableBuilder"/>.
+        /// </summary>
+        /// <remarks>
+        /// Using this method can avoid latency and potential deadlocks due to the internal 
+        /// <see cref="IAmazonDynamoDB.DescribeTable(string)"/> call that is used to populate 
+        /// the SDK's cache of table metadata. It requires that the table's index schema described accurately, 
+        /// otherwise exceptions may be thrown and/or the results of certain DynamoDB operations may change. 
+        /// It is recommended to test your application prior to using this in production code.
```

Comment:
This line sounds obvious and preachy to me. I think you did enough warning of the requirements in the previous sentence.
`It is recommended to test your application prior to using this in production code.`

Reply:
Yeah, I suppose that's most things we do :-D

Fixed in https://github.com/aws/aws-sdk-net/compare/74020f5167a3bf2e54dd53bdeba482f965e43dd2..abcc18e865203cd0aa124c2c175f0dc798d5be93.